app/services/folio_client.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require 'http'
+
+class FolioClient
+  DEFAULT_HEADERS = {
+    accept: 'application/json, text/plain',
+    content_type: 'application/json'
+  }.freeze
+
+  attr_reader :base_url
+
+  def initialize(url: ENV.fetch('OKAPI_URL'),
+                 username: ENV.fetch('OKAPI_USER', nil),
+                 password: ENV.fetch('OKAPI_PASSWORD', nil),
+                 tenant: 'sul')
+    uri = URI.parse(url)
+
+    @base_url = url
+    @username = username
+    @password = password
+
+    if uri.user
+      @username ||= uri.user
+      @password ||= uri.password
+      @base_url = uri.dup.tap do |u|
+        u.user = nil
+        u.password = nil
+      end.to_s
+    end
+
+    @tenant = tenant
+  end
+
+  def get(path, **kwargs)
+    authenticated_request(path, method: :get, **kwargs)
+  end
+
+  def get_json(path, **kwargs)
+    parse(get(path, **kwargs))
+  end
+
+  def post(path, **kwargs)
+    authenticated_request(path, method: :post, **kwargs)
+  end
+
+  def post_json(path, **kwargs)
+    parse(post(path, **kwargs))
+  end
+
+  # FOLIO Edge API - Real Time Availability Check
+  # https://s3.amazonaws.com/foliodocs/api/edge-rtac/p/edge-rtac.html
+  def rtac(instance_ids:, full_periodicals: true)
```

Comment:
Can we call this `real_time_availability`?

Reply:
yes, sounds good